x-pack/plugins/observability_solution/infra/server/lib/helpers/get_infra_metrics_client.test.ts

```diff
@@ -0,0 +1,171 @@
+import { InfraPluginRequestHandlerContext } from '../../types';
+import { QueryDslQueryContainer } from '@kbn/data-views-plugin/common/types';
+
+const withExcludedDataTiers = (tiers: DataTier[]) => ({
+  uiSettings: {
+    client: {
+      get: () => Promise.resolve(tiers),
+    },
+  },
+});
+
+const mockedInfra = { getMetricsIndices: () => Promise.resolve(['*.indices']) };
+
+const infraMetricsTestHarness =
+  (input: QueryDslQueryContainer | undefined, output: QueryDslQueryContainer) => async () => {
+    const callWithRequest = jest.fn();
+
+    const mockedCore = withExcludedDataTiers(['data_frozen']);
+
+    const context = {
+      infra: Promise.resolve(mockedInfra),
+      core: Promise.resolve(mockedCore),
+    } as unknown as InfraPluginRequestHandlerContext;
+
+    const client = await getInfraMetricsClient({
+      libs: { framework: { callWithRequest } } as unknown as InfraBackendLibs,
+      context,
+      request: {} as unknown as KibanaRequest,
+    });
+
+    await client.search({
+      body: {
+        query: input,
+        size: 1,
+        track_total_hits: false,
+      },
+    });
+
+    expect(callWithRequest).toBeCalledWith(
+      context,
+      'search',
+      {
+        body: {
+          query: output,
+          size: 1,
+          track_total_hits: false,
+        },
+        ignore_unavailable: true,
+        index: ['*.indices'],
+      },
+      {}
+    );
+  };
+
+describe('getInfraMetricsClient', () => {
+  it(
+    'includes excluded data tiers in the request filter by default',
+    infraMetricsTestHarness(undefined, {
+      bool: {
+        filter: [
+          {
+            bool: {
+              must_not: [
+                {
+                  terms: {
+                    _tier: ['data_frozen'],
+                  },
+                },
+              ],
+            },
+          },
+        ],
+      },
+    })
+  );
+
+  it(
+    'merges provided filters with the excluded data tier filter',
+    infraMetricsTestHarness(
+      {
+        bool: {
+          filter: [
+            {
+              bool: {
+                must: {
+                  exists: {
+                    field: 'a-field',
+                  },
+                },
+              },
+            },
+          ],
+        },
+      },
+      {
+        bool: {
+          filter: [
+            {
+              bool: {
+                must: {
+                  exists: {
+                    field: 'a-field',
+                  },
+                },
+              },
+            },
+            {
+              bool: {
+                must_not: [
+                  {
+                    terms: {
+                      _tier: ['data_frozen'],
+                    },
+                  },
+                ],
+              },
+            },
+          ],
+        },
+      }
+    )
+  );
+
+  it(
+    'merges other query params with the excluded data tiers filter',
+    infraMetricsTestHarness(
+      {
```

Comment:
I don't think `query` can have multiple `bool` objects: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-bool-query.html.

Did you test this on the devtools?

Reply:
I redid the object merging and confirmed on the dev tools that they are good.